docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,226 @@
+- object literals with "simple literals" used as keys and values (again, spread operators are ignored)
+
+## Dependency Analysis
+
+UI5 Tooling extracts dependency information from a project's code as outlined in the following sections.
+The APIs described there require the usage of "simple literals" when declaring dependencies.
+
+### JSModule Analyzer
+
+The following APIs are analyzed by the JSModule Analyzer:
+
+- sap.ui.define
+- sap.ui.require
+- jQuery.sap.declare (deprecated)
+- jQuery.sap.require (deprecated)
+- sap.ui.requireSync (deprecated)
+- sap.ui.preload (restricted)
+- sap.ui.require.preload (restricted)
+
+The [JSModule Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/JSModuleAnalyzer.js) uses the ["Abstract Syntax Tree"](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (AST) of a JavaScript file to decide whether a code block is executed *conditionally* or *unconditionally*.
+
+The analyzer uses a set of rules to decide whether one of the above APIs is called whenever the module is executed or whether the API is only called under certain conditions.
+
+For example, top-level code is always executed. Flow-control statements in JavaScript imply that certain blocks of code are only executed under certain conditions (for example, `if` blocks, `else` blocks, ...). Besides these inherent JavaScript rules, further common patterns are known to the analyzer, e.g. immediately invoked function expressions or the factor function of AMD modules.
+
+Any dependencies found that - according to these rules - are always executed, are collected as eager (or standard) dependencies. Dependencies that are found on a code path that depends on certain conditions are collected as conditional dependencies.
+
+The bundling implemented by UI5 Tooling can either follow only eager dependencies (resolve:`true`) or additionally conditional dependencies (resolveConditional). For more information, see [Custom Bundling](https://sap.github.io/ui5-tooling/v3/pages/Configuration/#custom-bundling).
+
+When a dependency in one of the mentioned APIs is not a "simple literal" but an expression, the corresponding module is marked as "having dynamic dependencies". This marker is currently not further evaluated by the tooling.
+
+### Component Analyzer
+
+The [Component Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/ComponentAnalyzer.js) analyzes JavaScript files named `Component.js` to collect dependency information by searching for a `manifest.json` in the same folder. If one is found, the `sap.ui5` section is evaluated in the following way:
+
+- Any library dependency is added as a dependency to the `library.js` module of that library. If the library dependency is modeled as 'lazy', the
+ module dependency will be added as 'conditional',
+- any component dependency is added as a dependency to a file named `Component.js`. If the dependency is modeled as 'lazy', the module dependency will be added as 'conditional',
```

Comment:
@KlattG now some words starting with uppercase in lists and some in lowercase. Could you please have another look on them?

Reply:
Ah got it, you have structured the list as a sentence.

Reply:
That is actually fine, because the list consists of full sentences that continue over several list points. In this case the usual sentence capitalization should be used.

Reply:
Just learned something new :-) Thanks @KlattG

Reply:
I just had a look into the SAP Style Guide - they recommend to begin the first word of each list item with a letter in the same case. Since we have full sentences here, I'd propose to change all first words to uppercase in this list.